src/internals/scan.rs

```diff
@@ -64,23 +67,24 @@ pub fn rules_scan_mem<'a>(
     mem: &[u8],
     timeout: i32,
     flags: i32,
-) -> Result<Vec<Rule<'a>>, YaraError> {
-    let mut results = Vec::<Rule<'a>>::new();
+    callback: impl FnMut(CallbackMsg<'a>) -> CallbackReturn,
```

Comment:
Is it possible to alias the callback type? It is used multiple time.

Reply:
No, it is impossible, needed: https://doc.rust-lang.org/beta/unstable-book/language-features/trait-alias.html